tests/integration/test_charm.py

```diff
@@ -35,20 +35,36 @@ async def test_build_and_deploy(ops_test, charm_under_test):
 
 async def test_relate(ops_test):
     await asyncio.gather(
-        ops_test.model.add_relation(PROM_NAME, APP_NAME),
-        ops_test.model.add_relation(APP_NAME, SPRING_NAME),
+        ops_test.model.add_relation(
+            f"{PROM_NAME}:metrics-endpoint", f"{APP_NAME}:metrics-endpoint"
+        ),
+        ops_test.model.add_relation(
+            f"{APP_NAME}:configurable-scrape-jobs", f"{ZINC_NAME}:metrics-endpoint"
+        ),
     )
     await ops_test.model.wait_for_idle(status="active")
 
 
 async def test_alert_rules_exist(ops_test):
     rules = await get_prometheus_rules(ops_test=ops_test, app_name=PROM_NAME, unit_num=0)
-    assert len(rules) > 0, "No alert rules are present even though spring music is related"
+    assert len(rules) > 0, "No alert rules are present even though zinc is related"
 
 
 async def test_multiple_workloads_alert_rules(ops_test):
     old_rules = await get_prometheus_rules(ops_test=ops_test, app_name=PROM_NAME, unit_num=0)
-    await ops_test.model.add_relation(APP_NAME, "spring-music2")
+    await ops_test.model.add_relation(APP_NAME, "zinc2")
     await ops_test.model.wait_for_idle(status="active")
     new_rules = await get_prometheus_rules(ops_test=ops_test, app_name=PROM_NAME, unit_num=0)
     assert len(new_rules) > len(old_rules), "Additional workload instance did not add alert rules"
+
+
+async def test_non_leader_units_set_waiting_status(ops_test):
```

Comment:
Can this be converted to a unit test?
I would also test if a non leader unit recovers properly into a leader, e.g. if unit/1 is leader and scaling down.

Reply:
Typically we check for statuses in utests, not itests.

Reply:
In this case there is a subtle difference doing this in a integration test and a unit test. A harness based unit test can not be used to check the status of two different units of the *same application*, at the *same time* , under the *same pre-conditions*.

Reply:
Isn't it something than can be conveniently captured with
```python
for is_leader in [True, False]:
    with self.subTest(is_leader=is_leader):
        # test goes here 
        # can add units and update peer rel data if needed
```
Just trying to move as much as possible to utests. Leaving for your judgement.

Reply:
I prefer unit tests too but subtest also constitutes separate tests. Harness is designed (for good reason I am sure) to test a single unit of a single charm by sending it messages through mocked relations etc. and there is no way to get around this fact other than an integration test.